torch/nn/modules/conv.py

```diff
@@ -460,35 +464,54 @@ def forward(self, input, output_size=None):
         else:
             return func(input, self.weight, self.bias)
 
-    def _output_padding(self, input, output_size):
-        if output_size is None:
-            return self.output_padding
-
-        output_size = list(output_size)
-        k = input.dim() - 2
-        if len(output_size) == k + 2:
-            output_size = output_size[2:]
-        if len(output_size) != k:
-            raise ValueError(
-                "output_size must have {} or {} elements (got {})"
-                .format(k, k + 2, len(output_size)))
-
-        def dim_size(d):
-            return ((input.size(d + 2) - 1) * self.stride[d] -
-                    2 * self.padding[d] + self.kernel_size[d])
-
-        min_sizes = [dim_size(d) for d in range(k)]
-        max_sizes = [min_sizes[d] + self.stride[d] - 1 for d in range(k)]
-        for size, min_size, max_size in zip(output_size, min_sizes, max_sizes):
-            if size < min_size or size > max_size:
-                raise ValueError((
-                    "requested an output size of {}, but valid sizes range "
-                    "from {} to {} (for an input of {})").format(
-                        output_size, min_sizes, max_sizes, input.size()[2:]))
+    @weak_script_method
+    def to_list(self, elem):
+        # type: (BroadcastingList1[int]) -> List[int]
+        return elem
 
-        return tuple([output_size[d] - min_sizes[d] for d in range(k)])
+    @weak_script_method
+    def _output_padding(self, input, output_size, stride, padding, kernel_size):
```

Comment:
is the function signature change necessary?

Reply:
The alternative is to make lists a constant of the module. By changing the signature, the tuples get converted to lists upon entry of the function, and we can index into the lists with a non-constant value